### ISSUE TITLE
Add wiki page to document markdown flavor + extensions

### ISSUE THREAD

**Issue by nagledb** [BoostIO/Boostnote#2421]
Problem: There are many flavors/variants of the Markdown syntax. There are also many extensions to Markdown. I am currently unable to find any documentation for Boostnote that addresses which variant of Markdown it supports and which extensions it might include.

Proposed solution:  On the wiki, add a page under the "Usage" section named "Markdown support" or similar that provides this information.

Skimming through the code, it looks like you're using [markdown-it](https://github.com/markdown-it/markdown-it) to parse and render markdown. Is that correct? Are the default options/rules being used or are any modifications made? Are any of its syntax extensions enabled? Is there any way for an end-user to add additional syntax extensions? Those are the kinds of questions I'd love to see answered on the wiki page.

**Comment by ZeroX-DG** [comment]
Yes, we're using markdown-it to parse and render the markdown. Currently, you can only add custom syntax by submitting a PR containing your plugin for markdown-it.
